### PR TITLE
Added github authentication & README CRUD operations

### PR DESCRIPTION
**Pull request by kdsolo**
resolves #159 
Also added github authentication (still need access control).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,27 +23,119 @@ https://docs.mongodb.com/manual/tutorial/install-mongodb-on-os-x/ (OS X)
...
+```
+> db.collection.insert()
+> db.collection.insertOne()
+> db.collection.insertMany()
+```
+
+Read operations to retrieve documents from a collection:
+
+```
+> db.collection.find()
```
Comment (sfoster) [inline README.md:82]: Can you add an example of a predicate here - like show all users whose status value is 'on'?

File: src/server/index.js [hunk 2]
```diff
@@ -16,13 +22,41 @@ var app = express();
...
+store.on('error', function(error) {
+  assert.ifError(error);
+  assert.ok(false);
+});
 
+if (process.env.NODE_ENV == 'production') {
+passport.use(new githubStrategy({
+    clientID: gitInfo.info.GITHUB_CLIENT_ID,
+    clientSecret: gitInfo.info.GITHUB_CLIENT_SECRET,
+    callbackURL: "http://127.0.0.1:3000/auth/github/callback"
```
Comment (sfoster) [inline src/server/index.js]: Should be process.env.PORT or config.port?

Comment (kdsolo) [reply src/server/index.js]: yes thank you

File: README.md (hunk 1 above)
Comment (kdsolo) [reply README.md:82]: Done

File: package.json [hunk 3]
```diff
@@ -5,11 +5,15 @@
...
     "body-parser": "^1.15.2",
+    "connect-mongodb-session": "^1.1.1",
     "cors": "^2.7.1",
     "express": "^4.14.0",
+    "express-session": "^1.14.0",
     "mongodb": "^2.2.6",
     "mongoose": "^4.5.9",
     "optimist": "^0.6.1",
+    "passport": "^0.3.2",
+    "passport-github": "^1.1.0",
```
Comment (sfoster) [inline package.json]: the passport docs link to passport-github2 as the current implementation, do we want to use that instead?

Comment (kdsolo) [reply package.json]: Ok changed to passport-github2.

**Comment by sfoster** [comment]
Looks good, thanks

**Pull request merged**